src/lib/one-graph/client.js

```diff
@@ -0,0 +1,835 @@
+
+const ONEDASH_APP_ID = '0b066ba6-ed39-4db8-a497-ba0be34d5b2a'
+
+const httpOkLow = 200
+const httpOkHigh = 299
+
+const basicPost = async (url, options) => {
+  const reqBody = options.body || ''
+  const userHeaders = options.headers || {}
+
+  const headers = {
+    ...userHeaders,
+    'Content-Type': 'application/json',
+    'Content-Length': reqBody.length,
+  }
+
+  const timeoutMilliseconds = 30_000
+
+  const resp = await fetch(url, {
+    method: 'POST',
+    headers,
+    timeout: timeoutMilliseconds,
+    compress: true,
+    body: reqBody,
+  })
+
+  const respBody = await resp.text()
+
+  if (resp.status < httpOkLow || resp.status > httpOkHigh) {
+    console.warn('Response:', respBody)
+    throw new Error(`Netlify OneGraph return non - OK HTTP status code: ${resp.status}`)
+  }
+
+  return respBody
+}
+
+const fetchOneGraphSchemaJson = async (appId, enabledServices) => {
+  const url = `https://serve.onegraph.com/schema?app_id=${appId}&services=${enabledServices.join(',')}`
+  const headers = {}
+
+  try {
+    const response = await basicPost(url, {
+      method: 'GET',
+      headers,
+      body: null,
+    })
+
+    return JSON.parse(response)
+  } catch (error) {
+    console.error('Error fetching schema:', error)
+  }
+}
+
+const fetchOneGraphSchema = async (appId, enabledServices) => {
+  const result = await fetchOneGraphSchemaJson(appId, enabledServices)
+  const schema = buildClientSchema(result.data)
+  return schema
+}
+
+const fetchOneGraph = async (
+  accessToken,
+  appId,
+  query,
+  operationName,
+  variables,
+  // eslint-disable-next-line max-params
+) => {
+  const payload = {
+    query,
+    variables,
+    operationName,
+  }
+
+  const body = JSON.stringify(payload)
+  try {
+    const result = await basicPost(`https://serve.onegraph.com/graphql?app_id=${appId}&show_metrics=false`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        Authorization: accessToken ? `Bearer ${accessToken}` : '',
+      },
+      body,
+    })
+
+    // @ts-ignore
+    const value = JSON.parse(result)
+    if (value.errors) {
+      console.warn(`${NETLIFYDEVWARN} fetchOneGraph errors`, operationName, JSON.stringify(value, null, 2))
+    }
+    return value
+  } catch {
+    return {}
+  }
+}
+
+const fetchOneGraphPersisted = async (
+  appId,
+  accessToken,
+  docId,
+  operationName,
+  variables,
+  // eslint-disable-next-line max-params
+) => {
+  const payload = {
+    doc_id: docId,
+    variables,
+    operationName,
+  }
+  try {
+    const result = await basicPost(`https://serve.onegraph.com/graphql?app_id=${appId}`, {
+      method: 'POST',
+      headers: {
+        Authorization: accessToken ? `Bearer ${accessToken}` : '',
+      },
+      body: JSON.stringify(payload),
+    })
+
+    return JSON.parse(result)
+  } catch {
+    return {}
+  }
+}
+
+const internalOperationsDoc = `mutation CreatePersistedQueryMutation(
```

Comment:
for readability I would suggest moving this mutation to it's own file as it makes reading the code harder

Reply:
Yes, all queries and mutations would be best defined in another file

Reply:
Why? I don't understand this requirement.

Reply:
I've moved it into a separate file, but now the definition for the graphql source and the function definition are in two separate files. This seems like a regression to me.